server.py

```diff
@@ -141,6 +141,9 @@ def check_content_type(content_type):
 @app.route('/shopcarts', methods=['GET'])
 def get_all_shopcarts():
     carts = Shopcart.all()
```

Comment:
This should probably be within an if/else with the `find_by_product`, so you don't always get all carts regardless if you're going to query by product.

Reply:
Actually you want to get all because you are in the GET /shopcarts
so you only filter if you have a query string.

Reply:
But I see  your point and I agree, I followed the code from rofrano. haha